Integrations/integration-ThreatX.yml

```diff
@@ -0,0 +1,728 @@
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: ThreatX Server URL (e.g., https://provision.threatx.io)
+  name: url
+  defaultvalue: https://provision.threatx.io
+  type: 0
+  required: true
+- display: API Key
+  name: api_key
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: DBot Score Threshold
+  name: dbot_threshold
+  defaultvalue: "70"
+  type: 0
+  required: true
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+- display: Trust any certificate (not secure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    ''' IMPORTS '''
+    import socket
+    import struct
+    import time
+    from operator import itemgetter
+    import requests
+
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    ''' GLOBAL VARS '''
+    CUSTOMER_NAME = demisto.params().get('customer_name', None)
+    API_KEY = demisto.params().get('api_key', None)
+    URL = demisto.params().get('url', None)
+
+    if URL[-1] != '/':
+        URL += '/'
+
+    BASE_URL = URL + 'tx_api/v1'
+    DBOT_THRESHOLD = int(demisto.params().get('dbot_threshold', 70))
+    USE_SSL = not demisto.params().get('insecure')
+
+    ''' HELPER FUNCTIONS '''
+    def http_request(url_suffix, commands=None):
+        state = demisto.getIntegrationContext()
+
+        if not state.get('session_token'):
+            session_token = None
+            token_expires = None
+        else:
+            session_token = state.get('session_token')
+            token_expires = state.get('token_expires')
+
+        if url_suffix != '/login':
+            demisto.info('running request with url=%s with commands=%s' % (BASE_URL + url_suffix, commands))
+            data = {
+                'token': session_token,
+                'customer_name': CUSTOMER_NAME
+            }
+        else:
+            demisto.info('running request with url=%s' % (BASE_URL + url_suffix))
+            data = {}
+
+        if commands is not None:
+            data.update(commands)
+
+        res = requests.request('POST',
+                               BASE_URL + url_suffix,
+                               verify=USE_SSL,
+                               json=data)
+
+        if res.status_code != requests.codes.ok:
+            if url_suffix == '/login':
+                demisto.setIntegrationContext({'session_token': None,
+                                               'token_expires': None
+                                               })
+                demisto.info(str(res.status_code) + ' from server during login. Clearing session token cache.')
+
+            return_error('HTTP %d Error in API call to ThreatX service - %s' % (res.status_code, res.text))
+
+        if not res.text:
+            resp_json = {}
+
+        try:
+            resp_json = res.json()
+        except:
+            return_error('Could not parse the response from ThreatX.')
```

Comment:
Please add `res.content` or `res.text` to the error message, if they include additional details

Reply:
done